src/callbacks.jl

```diff
@@ -45,3 +32,202 @@ function reeval_internals_due_to_modification!(integrator::DDEIntegrator)
 
     integrator.u_modified = false
 end
+
+# track propagated discontinuities for dependent delays
+
+struct DiscontinuityCallback{F,D<:Discontinuity,A,R} <: AbstractContinuousCallback
+    lags::F
+    discontinuities::Vector{D}
+    interp_points::Int
+    atol::A
+    rtol::R
+end
+
+"""
+    DiscontinuityCallback(lags, discontinuities::Vector{<:Discontinuity};
+                          [interp_points::Int=10, atol=1e-12, rtol=0])
+
+Callback that tracks `discontinuities` that are propagated by dependent `lags` of the form
+`(t, u) -> lag(t, u)`.
+
+Hereby a number `interp_points` of interpolation points are used to first check for
+different signs of functions ``f(t) = T + lag(t, u(t)) - t``, where ``T`` is time point of a
+previous discontinuity and ``t`` is contained in the current time interval. This shows that
+the current time interval contains propagated discontinuities of which the exact time point
+is then determined by a root finding algorithm. The sign at the lower bound of the time
+interval, i.e. at `tprev`, is set to 0 with absolute tolerance `atol` and relative
+tolerance `rtol`.
+"""
+function DiscontinuityCallback(lags, discontinuities::Vector{<:Discontinuity};
+                               interp_points::Int=10, atol=1e-12, rtol=0)
+    DiscontinuityCallback(lags, discontinuities, interp_points, atol, rtol)
+end
+
+# do not initialize discontinuity callback
+initialize!(t, u, integrator::DEIntegrator, c::DiscontinuityCallback) = nothing
+
+# find time of first discontinuity in the current time interval (if existent)
+function find_callback_time(integrator::DDEIntegrator, callback::DiscontinuityCallback)
+    # initialize time and order of first discontinuity in the current time interval
+    tmin = zero(integrator.t)
+    order = 0
+
+    # update interpolation data and calculate interpolation points
+    if callback.interp_points != 0
+      ode_addsteps!(integrator)
+    end
+    Θs = linspace(0, one(integrator.t), callback.interp_points)
+
+    # use temporary array of mutable caches in calculation of interpolants
+    if typeof(integrator.cache) <: OrdinaryDiffEq.OrdinaryDiffEqMutableCache
+        idxs_internal = eachindex(integrator.cache.tmp)
```

Comment:
don't use this for indexing. Always `idxs_internal = nothing`.

Reply:
Ah, I see, I mixed something up when I copied and simplified `callbacks.jl` from OrdinaryDiffEq.

Reply:
This was recently fixed.